tests/unit/report/data_sources/hubspot_client_test.py

```diff
@@ -89,6 +89,21 @@ def test_get_associations(self, client, batch_response, BatchInputPublicObjectId
             (result._from.id, result.to[1].id),
         }
 
+    def test_get_associations_batches_calls(self, client, BatchInputPublicObjectId):
+        client.get_associations(
+            from_type=client.AssociationObjectType.COMPANY,
+            to_type=client.AssociationObjectType.DEAL,
+            object_ids=list(range(client.ASSOCIATIONS_BATCH_SIZE + 10)),
+        )
+        BatchInputPublicObjectId.assert_has_calls(
+            [
+                call(inputs=Any.list.of_size(client.ASSOCIATIONS_BATCH_SIZE)),
+                call(inputs=Any.list.of_size(10)),
+            ]
+        )
+
+        assert client.api_client.crm.associations.batch_api.read.call_count == 2
```

Comment:
The main functionality is tested above, we are just testing it batches things here.